A4A > Marketplace: Update the standard hosting slider to support input values

yashwin opened the pull request:
Closes https://github.com/Automattic/jetpack-genesis/issues/465

## Proposed Changes

* This PR updates the standard agency hosting slider to support input values.

## Why are these changes being made?

* To allow users to interact with the slider and update the slider according to the input values and vice-versa. 

## Testing Instructions

* Open the A4A live link
* Go to Marketplace - Hosting - Standard agency hosting > Interact with the slider at various scenarios and verify it works as expected. Interacting with both slider and input box should update the values accordingly.

1) When the user has 0 plans


https://github.com/user-attachments/assets/30ca7931-a953-4cb8-b1e8-e41ffc4625ae


2) When the user has 1-9 plans

https://github.com/user-attachments/assets/9725cac5-97ba-4f8c-bf6c-dd02d6550433


3) When the user has > 10 plans 

https://github.com/user-attachments/assets/d60a7472-6d6b-4172-98c2-413c2347f84a


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

yashwin committed:
Update the standard hosting slider

cleacos reviewed (changes requested):
I check it on Horizon, and it doesn't work well for me:

<img width="554" alt="image" src="https://github.com/user-attachments/assets/2b2cdda0-e6f6-45c4-9ed0-3edb1661e37f">

- The pointer is in the middle of 1-3 instead of 1
- The slider doesn't move step by step: 1 -> 3 -> 4, it moves continuosly and stops at any point.

yashwin commented:
> I check it on Horizon, and it doesn't work well for me:

Thanks for the review, @cleacos! 

> * The pointer is in the middle of 1-3 instead of 1

This is correct, as users can purchase 2 licenses. Since there is no discount for 2 plans, we don't have a marker for that.

> * The slider doesn't move step by step: 1 -> 3 -> 4, it moves continuosly and stops at any point.

This is also correct. Since the users can purchase 2, I think we allow it to free flow rather than forcing them to select only the available options as in Pressable. 

This is how it works in production currently: 

<img width="1073" alt="Screenshot 2024-08-06 at 8 17 09 AM" src="https://github.com/user-attachments/assets/93fa6854-026d-41e5-ab68-6c1ca585b3dc">

madebynoam commented:
> This is also correct. Since the users can purchase 2, I think we allow it to free flow rather than forcing them to select only the available options as in Pressable.


@yashwin is hard to make it snap to the values? I know that's what we currently have, but feels a bit broken this way :) 
If it makes it easier to fix, we could just display the 2 value without a discount (like we do for 1).

madebynoam commented:
At around tablet size breakpoints, we get a funny looking wrapping. Maybe at that size, we should break the discount to its own line? If possible?
<img width="372" alt="image" src="https://github.com/user-attachments/assets/4406a562-33f7-4508-9dc1-cd9d4fea5ada">

madebynoam commented:
Other than those 2 issues, looks good to me!

yashwin committed:
address PR comments

yashwin commented:
Thanks for the review, @madebynoam!

> If it makes it easier to fix, we could just display the 2 value without a discount (like we do for 1).

I added the option for 2 plans [now](https://github.com/Automattic/wp-calypso/pull/93239/commits/3911c3de9093dfec5a07c9d8b1598dac9ab4e1f0). But the UX is not the same as Pressable. These are 2 different components. I'm not sure how easy it is to fix now. Maybe we can address this in a different PR post release. WDYT?

> At around tablet size breakpoints, we get a funny looking wrapping. Maybe at that size, we should break the discount to its own line? If possible?

Fixed it now: https://github.com/Automattic/wp-calypso/pull/93239/commits/3911c3de9093dfec5a07c9d8b1598dac9ab4e1f0

madebynoam commented:
> I added the option for 2 plans [now](https://github.com/Automattic/wp-calypso/pull/93239/commits/3911c3de9093dfec5a07c9d8b1598dac9ab4e1f0). But the UX is not the same as Pressable. These are 2 different components. I'm not sure how easy it is to fix now. Maybe we can address this in a different PR post release. WDYT?

Sounds good, let's plan to circle back on this one, @yashwin.
CC @jeffgolenski

madebynoam commented:
@yashwin Could you create a maintenance issue for the slider? 🙌

cleacos commented:
I mean, the slider is continuous, I can leave the pointer in the middle between 2 and 3, instead of being sticky to the points:

![image](https://github.com/user-attachments/assets/ae33f307-8067-4ac1-bd4e-41f6c7a8fb55)

A minor UI issue, the point is not alignment with the number spot:

![image](https://github.com/user-attachments/assets/ea78fc63-d3f6-49d3-b599-c227b9fddc5b)

cleacos commented:
Sorry, I was checking the slider in production... I don't know but I thought the slider point was sticky (magnet) to each point...

cleacos commented:
OK, I was not wrong at all... The slider I had in my mind on the product page, and if you move it, it is sticky to each point:

![image](https://github.com/user-attachments/assets/d89d3b27-81f9-499f-a75d-4b49463e8563)

yashwin commented:
@madebynoam:

Yes, I have created one here: https://github.com/Automattic/jetpack-genesis/issues/468

@cleacos:

This is an existing issue, and we will tackle this separately in a different PR. [Ticket](https://github.com/Automattic/jetpack-genesis/issues/468)

cleacos reviewed (approved):
Cool!

LGTM!
